src/wallet.rs

```diff
@@ -0,0 +1,962 @@
+        Some(String::from(name))
+    }
+
+    /// Returns current directory for this path
+    pub fn dir(&self) -> Option<PathBuf> {
+        let mut p = self.0.clone();
+        if p.pop() {
+            Some(p)
+        } else {
+            None
+        }
+    }
+
+    /// Sets the directory for the state cache
+    pub fn set_cache_dir(path: &Path) -> Result<(), Error> {
+        State::set_cache_dir(path.to_path_buf()).map_err(Error::from)
+    }
+
+    /// Wallet name defaults to user's username
+    pub fn default_name() -> String {
+        // get default user as default wallet name (remove whitespace)
+        let mut user: String = whoami::username();
+        user.retain(|c| !c.is_whitespace());
+        user
+    }
+
+    /// Wallet default directory defaults to user's home directory
+    pub fn default_dir() -> PathBuf {
+        let home = dirs::home_dir().expect("OS not supported");
+        Path::new(home.as_os_str()).join(DATA_DIR)
+    }
+
+    /// Checks if a wallet with this name already exists
+    pub fn exists(dir: &Path, name: &str) -> bool {
+        let mut pb = dir.to_path_buf();
+        pb.push(name);
+        pb.set_extension("dat");
+        pb.is_file()
+    }
+
+    /// Get full paths of all wallet files found in `dir`
+    pub fn wallets_in(dir: &Path) -> Result<Vec<WalletPath>, Error> {
+        let dir = fs::read_dir(dir)?;
+        let wallets = dir
+            .filter_map(|el| el.ok().map(|d| d.path()))
+            .filter(|path| path.is_file())
+            .filter(|path| match path.extension() {
+                Some(ext) => ext == "dat",
+                None => false,
+            })
+            .map(WalletPath)
+            .collect();
+
+        Ok(wallets)
+    }
+}
+
+impl Default for WalletPath {
+    fn default() -> Self {
+        let mut pb = Self::default_dir();
+        pb.push(Self::default_name());
+        pb.set_extension("dat");
+        WalletPath(pb)
+    }
+}
+
+impl FromStr for WalletPath {
+    type Err = crate::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let p = Path::new(s);
+        Ok(Self(p.to_owned()))
+    }
+}
+
+impl From<PathBuf> for WalletPath {
+    fn from(p: PathBuf) -> Self {
+        Self(p)
+    }
+}
+
+impl From<&Path> for WalletPath {
+    fn from(p: &Path) -> Self {
+        Self(p.to_owned())
+    }
+}
+
+impl Display for WalletPath {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0.display())
+    }
+}
+
+#[derive(Debug)]
+/// Gas price and limit for any transaction
+pub struct Gas {
+    price: Lux,
+    limit: u64,
+}
+
+impl Gas {
+    /// Default gas price and limit
+    pub fn new() -> Self {
+        Gas {
+            price: DEFAULT_GAS_PRICE,
+            limit: DEFAULT_GAS_LIMIT,
+        }
+    }
+
+    /// Create gas from parsed runtime arguments
+    pub fn from_args(price: Option<Lux>, limit: Option<u64>) -> Self {
```

Comment:
The documentation and the method name are misleading, since we're in a library, there is no reference to "runtime arguments".

IMVHO, we should just remove this method; if the consumer wants to have a facility function because this happens a lot (but I don't think so), it can create its own:
```rust
// we're assuming we're in the binary here, not the library
fn gas_from_args(price: Option<Lux>, limit: Option<u64>) {
     let mut gas = Gas::new();
    
    if let Some(value) = price {
        gas.set_price(value);
    }
    if let Some(value) = limit {
        gas.set_limit(value)
    }
    gas
}

Reply:
It is used 4 times, thus I felt it was convenient, but I see your point. Will move it to the binary.